### PR TITLE
Safe distance label lateral

### PR DESCRIPTION
**Pull request by juloberno**
- Label Safe Distance: now includes option to calculate lateral safe distance
- GetFrontRear now truly considers lateral difference offset with option to restrict it to lane polygon or not
- Fixes in FrenetState and SingleTrackLaneChange controller (not working for intersection scenario)
- Fix wrapping of primitives ConstAccStayLane

### CODE REVIEW

File: bark/models/behavior/rule_based/lane_change_behavior.cpp [hunk 1]
```diff
@@ -50,7 +50,11 @@ BehaviorLaneChangeRuleBased::FrontRearAgents(
     const ObservedWorld& observed_world,
     const LaneCorridorPtr& lane_corr) const {
   AgentInformation front_info, rear_info;
-  const auto& front_rear = observed_world.GetAgentFrontRear(lane_corr);
+  double lateral_difference_threshold = 100; // do not restrict regarding lateral diff
```
[the remark this answers is not in the document]
Comment (juloberno) [reply bark/models/behavior/rule_based/lane_change_behavior.cpp:53]: yes

[the remark this answers is not in the document]
Comment (patrickhart) [reply bark/models/behavior/rule_based/lane_change_behavior.cpp:53]: should this not be externally settable via the parameter server?

File: bark/models/dynamic/single_track.hpp [hunk 2]
```diff
@@ -158,7 +191,7 @@ inline double CalculateSteeringAngle(const SingleTrackModelPtr& model,
 
   FrenetState f_state = FrenetState(state_front, ref_line);
   double vel = state(StateDefinition::VEL_POSITION);
-  double delta = -NormToPI(f_state.angle) + atan2(-gain * f_state.lat, vel);
+  double delta = +NormToPI(f_state.angle) + atan2(-gain * f_state.lat, vel);
```
Comment (patrickhart) [inline bark/models/dynamic/single_track.hpp:194]: no need for a `+` sign

File: bark/runtime/viewer/viewer.py [hunk 3]
```diff
@@ -461,6 +461,7 @@ def drawLaneCorridor(self, lane_corridor, color=None):
             color = "blue"
         self.drawPolygon2d(lane_corridor.polygon, color=color,
                            facecolor=color, alpha=.3, zorder=2)
+        self.drawLine2d(lane_corridor.center_line, "k")
```
Comment (patrickhart) [inline bark/runtime/viewer/viewer.py:464]: could draw the center line in the same color as the lane corridor

File: bark/world/evaluation/ltl/label_functions/safe_distance_label_function.cpp [hunk 4]
```diff
@@ -162,10 +138,13 @@ bool SafeDistanceLabelFunction::CheckSafeDistance(
   double t_stop_f = -v_f / a_f;
```
Comment (patrickhart) [inline bark/world/evaluation/ltl/label_functions/safe_distance_label_function.cpp:138]: this assumes linear acceleration. would it not be better to use v=1/2*a*t^2

File: tools/debug/analyze_map.py [hunk 5]
```diff
@@ -20,13 +25,16 @@
...
+
+bark.core.commons.GLogInit("", "./", 5, True, "")
 
 import numpy as np
 import itertools
 
 # Name and Output Directory
 # CHANGE THIS #
-map_name = "DR_CHN_Merging_ZS_partial_v04_lower_edited"
+map_name = "left_turn_rural"
```
Comment (patrickhart) [inline tools/debug/analyze_map.py:37]: use map that is publicly available, so one can e.g. visualize the lane corridor and understand the concept

**Review by patrickhart — APPROVED** [review]

**Review by klemense1 — APPROVED** [review]

**Pull request merged**